pkg/transport/generic_transport_controller.go

```diff
@@ -0,0 +1,507 @@
+			if err != nil {
+				return nil, fmt.Errorf("failed to get required cluster-scoped object '%s' with gvr %s from WDS - %w", objectName, gvr, err)
+			}
+			objectsToPropagate = append(objectsToPropagate, cleanObject(object))
+		}
+	}
+	// add namespace-scoped objects to the 'objectsToPropagate' slice
+	for _, namespaceScopedObject := range placementDecision.Spec.Workload.NamespaceScope {
+		gvr := schema.GroupVersionResource{Group: namespaceScopedObject.Group, Version: namespaceScopedObject.Version, Resource: namespaceScopedObject.Resource}
+		dynamicClientGvrInterface := c.wdsDynamicClient.Resource(gvr)
+		for _, objectsByNamespace := range namespaceScopedObject.ObjectsByNamespace {
+			if objectsByNamespace.Names == nil {
+				continue // no objects from this namespace, skip
+			}
+			for _, objectName := range objectsByNamespace.Names {
+				object, err := dynamicClientGvrInterface.Namespace(objectsByNamespace.Namespace).Get(ctx, objectName, metav1.GetOptions{})
+				if err != nil {
+					return nil, fmt.Errorf("failed to get required namespace-scoped object '%s' in namespace '%s' with gvr '%s' from WDS - %w", objectName,
+						objectsByNamespace.Namespace, gvr, err)
+				}
+				objectsToPropagate = append(objectsToPropagate, cleanObject(object))
+			}
+		}
+	}
+
+	return objectsToPropagate, nil
+}
+
+func (c *genericTransportController) createOrUpdateWrappedObject(ctx context.Context, namespace string, wrappedObject *unstructured.Unstructured) error {
+	existingWrappedObject, err := c.transportClient.Resource(c.wrappedObjectGVR).Namespace(namespace).Get(ctx, wrappedObject.GetName(), metav1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) { // if object is not there, we need to create it. otherwise report an error.
+			return fmt.Errorf("failed to create wrapped object '%s' in destination WEC with namespace '%s' - %w", wrappedObject.GetName(), namespace, err)
+		}
+		// object not found when using get, create it
+		_, err = c.transportClient.Resource(c.wrappedObjectGVR).Namespace(namespace).Create(ctx, wrappedObject, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to create wrapped object '%s' in destination WEC mailbox namespace '%s' - %w", wrappedObject.GetName(), namespace, err)
+		}
+		return nil
+	}
+	// // if we reached here object already exists, try update object
+	wrappedObject.SetResourceVersion(existingWrappedObject.GetResourceVersion())
+	_, err = c.transportClient.Resource(c.wrappedObjectGVR).Namespace(namespace).Update(ctx, wrappedObject, metav1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update wrapped object '%s' in destination WEC mailbox namespace '%s' - %w", wrappedObject.GetName(), namespace, err)
+	}
+
+	return nil
+}
+
+// updateObjectFunc is a function that updates the given object. returns true if object was updated, otherwise false.
+type updateObjectFunc func(*v1alpha1.PlacementDecision) bool
```

Comment:
This type is not for _any_ object, only a specific kind of objects; I would name it something more specific like `updateDecisionFunc` to `updatePlacementDecisionFunc`. This is not a blocker.